bridges/darwinia-ethereum/bridge/src/service/extrinsics/mod.rs

```diff
@@ -30,31 +31,42 @@ impl Service for ExtrinsicsService {
         // Datastore
         let state = bus.storage().clone_resource::<BridgeState>()?;
 
+        let greet_state = state.clone();
         let _greet = Self::try_task(
             &format!("{}-service-extrinsics", DarwiniaEthereumTask::name()),
             async move {
-                let mut handler = ExtrinsicsHandler::new(state.clone()).await;
-
+                let handler = ExtrinsicsHandler::new(greet_state.clone()).await;
                 while let Some(recv) = rx.recv().await {
                     if let ToExtrinsicsMessage::Extrinsic(ex) = recv {
-                        while let Err(err) = handler.send_extrinsic(ex.clone()).await {
+                        if handler.collect_message(&ex).is_err() {
                             tracing::error!(
                                 target: "darwinia-ethereum",
-                                "extrinsics err: {:#?}",
-                                err
+                                "Failed to save extrinsic {:?}",
+                                &ex
                             );
-
-                            // TODO: Consider the errors more carefully
-
-                            tokio::time::sleep(std::time::Duration::from_secs(5)).await;
-                            handler = ExtrinsicsHandler::new(state.clone()).await;
                         }
                     }
                 }
+                Ok(())
+            }
+        );
 
+        let _consume = Self::try_task(
+            &format!("{}-service-extrinsics", DarwiniaEthereumTask::name()),
+            async move {
+                let handler = ExtrinsicsHandler::new(state.clone()).await;
+                if handler.consume_message().await.is_err() {
```

Comment:
We need to restart consume_message after the error. because we need to make sure this service is running